src/Chapter3.hs

```diff
@@ -753,6 +837,19 @@ parametrise data types in places where values can be of any general type.
   maybe-treasure ;)
 -}
 
+data TreasureChest x = TreasureChest
+    { treasureChestGold :: Int
+    , treasureChestLoot :: x
+    }
+
+data Dragon = MkDragon 
+  { dragonName :: String 
```

Comment:
This is correct, however, this allows dragon to have only the name (which is `String`) 🐲 

In the task description, we want it would be more:
> A dragon can have a unique magical power. But it can be literally anything!

So, you could add another field into this type with `dragonPower :: power` where `power` is polymorphic.

In that case, the `Lair` type would have 2 parameters as well 🙂